### ISSUE TITLE
Module Structure

### ISSUE THREAD

**Issue by wenkesj** [fomorians/pyoneer#2]
I'd like to outline a general idea for the future structure of this library:

+ `features` - see #1
+ `nn` - follows the same spirit of `tf.nn`, with activation functions and normalization:
  + `swish`
  + eager friendly`layer_normalization` similar to `batch_normalization`
  + `rnn_cell`, sub-module that follows same `RNNCell` tf api
    + `FastWeightsRNNCell` and `FastWeightsLSTMCell`
    + `TemporalConv1DCell`
+ `rl` - follows similar concepts as [trfl](https://github.com/deepmind/trfl), except higher-level tools for agents, like [dopamine](https://github.com/google/dopamine)
  + implementations of common RL operations, discounts, advantages
  + tools for efficient, high-level rollouts with gym and keras models.
  + `agents` - submodule with a set of abstract `Models` that implement `compute_loss`, `compute_gradients` and more that correspond to gradient estimators for RL algorithms. These are not abstract, baselines, they are templates for algorithms (see the general concepts introduced by the present code)
